doc/reference/reference_lua/box_space/create_index.rst

```diff
@@ -113,321 +124,404 @@ On this page:
+Using the path option with [*]
+--------------------------------------------------------------------------------
+
+The string in a path option can contain '[*]' which is called
+an array index placeholder. Indexes defined with this are useful
+for JSON documents that all have the same structure.
+
+For example, when creating an index on field#2 for a string document
+that will start with ``{'data': [{'name': '...'}, {'name': '...'}]``,
+the parts section in the create_index request could look like:
+``parts = {{field = 2, type = 'str', path = 'data[*].name'}}``.
+Then tuples containing names can be retrieved quickly with
+``index_object:select({key-value})``.
+
+In fact a single field can have multiple keys, as in this example
+which retrieves the same tuple twice because there are two keys 'A' and 'B'
+which both match the request:
+
+.. code-block:: lua
+
+    s = box.schema.space.create('json_documents')
+    s:create_index('primarykey')
+    i = s:create_index('multikey', {parts = {{field = 2, type = 'str', path = 'data[*].name'}}})
+    s:insert({1,
+             {data = {{name='A'},
+                      {name='B'}},
+              extra_field = 1}})
+    i:select({''},{iterator='GE'})
+
+The result of the select request looks like this:
+
+.. code-block:: tarantoolsession
+
+    tarantool> i:select({''},{iterator='GE'})
+    ---
+    - - [1, {'data': [{'name': 'A'}, {'name': 'B'}], 'extra_field': 1}]
+    - [1, {'data': [{'name': 'A'}, {'name': 'B'}], 'extra_field': 1}]
+    ...
+
+Some restrictions exist:
+
+* '[*]' must be alone or must be at the end of a name in the path
+* '[*]' must not appear twice in the path
+* if an index has a path with x[*] then no other index can have a path with
+  x.component
+* '[*]' must not appear in the path of a primary-key
+* if an index has ``unique=true`` and has a path with '[*]'
+  then duplicate keys from different tuples are disallowed but duplicate keys
+  for the same tuple are allowed
+* as with :ref:`Using the path option for map fields <box_space-path>`,
+  the field's value must have the structure that the path definition implies,
+  or be nil (nil is not indexed)
+
+.. _box_space-index_func:
+
+--------------------------------------------------------------------------------
+Making a functional index with space_object:create_index()
+--------------------------------------------------------------------------------
+
+Functional indexes are indexes that call a user-defined function for forming
+the index key, rather than depending entirely on the Tarantool default formation.
+Functional indexes are useful for condensing or truncating or reversing or
+any other way that users want to customize the index.
+
+The function definition must expect a tuple (which has the contents of
+fields at the time a data-change request happens) and must return a tuple
+(which has the contents that will actually be put in the index).
+
+The space must have a memtx engine.
```

Comment:
This section has a lot of one-sentence paragraphs. Maybe they can be joined into a few longer paragraphs or grouped in a list.

Reply:
> в [*]-секции хотели сделать маркированный список, но не получилось, надо исправить